README Translation (#24)
FriendsOfREDAXO/redaxo-mit-docker: CONTRIBUTING.md

# Get Involved

> Information about how this addon can be maintained and extended.

Reviewer: ```suggestion
Information about how this package can be maintained and extended.
```

## Edit the REDAXO Version

> 1. Edit the __ENV-Variables__.

Reviewer: ```suggestion
1. Edit the __ENV variables__.
```

    ENV REDAXO_VERSION=5.3.0 REDAXO_SHA=c1d3dcc2401b39cd9d32ea7de2ea358683ce81ceb8a5845f7a79937adcd5048e
    ```

>     :point_right: _The checksum of a REDAXO package you get in the console for example on this way:_

Reviewer: ```suggestion
    :point_right: _You can find out the checksum of a REDAXO package in the console e.g. like this:_
```

        $ curl -Ls https://github.com/redaxo/redaxo/releases/download/5.3.0/redaxo_5.3.0.zip | shasum

> 2. Check if `docker-redaxo.php` needs to be adjusted. Inside are the installation routines for REDAXO and Addons, some of which have been adapted from Core, Installer Add-on, and Backup Add-on.

Reviewer: ```suggestion
2. Check if `docker-redaxo.php` needs to be updated. It contains installation routines for REDAXO and addOns, which have been partially adapted from the core, the installer addOn and the backup addOn.
```

> 3. Check if `default.config.yml` needs to be adjusted.

Reviewer: ```suggestion
3. Check if `default.config.yml` needs to be updated.
```

## Adjust the demos

The configuration of the demos is located in `/docker/php-apache/demos.yml`.

> 1. __Download of the AddOns__

Reviewer: ```suggestion
1. __Download of addOns__
```

>      You can find information about which AddOns requires a demo in the `package.yml` of the respective demo. Note that REDAXO already brings some add-ons. You just have to install the add-ons that do not already exist.

Reviewer: ```suggestion
     You can find information about which addOns are required for a demo in the `package.yml` of the respective demo. Note that REDAXO already brings some addOns. You only have to install the addOns that are not already available.
```

>     Also, remember to list the demo itself as an add-on for download. Best right at the beginning of the list.

Reviewer: ```suggestion
    Also, remember to list the demo itself as an addOn for download. Best right at the beginning of the list.
```

>     :point_right: _The number for `file` is best found out by looking in the installer for the add-on. It is then in the URL._

Reviewer: ```suggestion
    :point_right: _The best way to find out the number for `file` is to search for the addOn in the installer. It will be located in the URL._
```

> 2. __Activation of addons and plugins__

Reviewer: ```suggestion
2. __Activation of addOns and plugIns__
```

>     The order of activation is very relevant because AddOns can have dependencies on each other! For example, the `phpmailer` is used by some AddOns, so it should be high on the list. The same applies to `yform`.

Reviewer: ```suggestion
    The order of activation is important because addOns may depend on each other! For example, since `phpmailer` is required by some addOns, it should be placed high up in the list. The same applies to `yform`.
```

>     At least, activate the Demo-Addon

Reviewer: ```suggestion
    The demo addOn should be the last addOn you activate!
```

> 3. __Database-Import__

Reviewer: ```suggestion
3. __Database import__
```

>     Check which SQL dumps have to be imported from the demo add-on.

Reviewer: ```suggestion
    Check which SQL dumps have to be imported from the demo addOn.
```

> 4. __File-Import__

Reviewer: ```suggestion
4. __File import__
```

>     Check which files need to be imported from the demo add-on.

Reviewer: ```suggestion
    Check which files need to be imported from the demo adOn.
```
